### ISSUE TITLE
Chrome 93 doesn't use the native file picker anymore

### ISSUE THREAD

**Issue by stack-not-available** [lxqt/lxqt#2069]
##### Expected Behavior

Normal, native file picker is displayed:

![native file picker](https://i.imgur.com/F6RgXTG.png)

##### Current Behavior

Non-native, maybe Gtk3 default file picker is displayed:

![native file picker](https://i.imgur.com/bBrDShQ.png)

Furthermore, when I download a file and click open inside Chrome (in the download bar, download page Ctrl+J), it open in a new tab (if Chrome can read it, like images), instead of using the system default app.

##### Possible Solution

No idea. The last Chrome build to exhibit the expected behavior is 93.0.4557.4, a snapshot build. Between that and the next build is quite a lot of commits, but I found [one commit][2] that might be related to this.

I originally [file a bug in chromium][1], but I was advised to move it here. Note the comment about `org.freedesktop.portal.Desktop.OpenFile`.

##### Steps to Reproduce (for bugs)

1. Save/open/... anything
2. Non-native file picker is displayed

##### Context

The current, simple file picker doesn't have a search functionality.

##### System Information

* Distribution & Version: 21.04 x64
* Kernel: 5.11.0-31-generic
* Qt Version: 5.15.2
* liblxqt Version: No idea, but should be latest
* Package version: 0.16.0

  [1]: https://bugs.chromium.org/p/chromium/issues/detail?id=1246806
  [2]: https://chromium-review.googlesource.com/c/chromium/src/+/2992214

**Comment by palinek** [comment]
I looked into the mentioned commit... it seems as we should implement the backend for https://github.com/flatpak/xdg-desktop-portal to forward the FileChooser requests to our file dialog...

**Comment by redtide** [comment]
hello all, I'm testing @yan12125' AUR package to make work a feature for an audio plugin project and I have some troubles to make it work. Looks like the installation needs `xdg-desktop-portal` (which IIUC is not optional, also on the repo' README is listed as required dependency). The problem seems that while installing an implementation, desktop portal requires it to be installed first (dependency cicle?), so I installed the kde impl and once installed the lxqt one, I removed the other, but I can't get the lxqt filedialog (and now not even the gtk one in firefox, that I run prefixed by `GTK_USE_PORTAL=1`).

**Comment by redtide** [comment]
> Is that audio plugin using Qt?

Nope, but as I said in the edited comment it worked after some tries, steps was:
1. install xdg-desktop-portal and xdg-desktop-portal-kde
2. install xdg-desktop-portal-lxqt
3. removed xdg-desktop-portal-kde and killed all its previous instances

So from there on I can open the basic Qt filedialog (see the screenshots in the link below) even with GTK applications (e.g. from Firefox), and from the audio plugin framework test application, that should use nanovg (so no Qt, nor GTK).
But it should be a LXQt filedialog instead.

I'm sure settings are OK here (including `QT_QPA_PLATFORMTHEME` that was set correctly [with the help of tsujan](https://github.com/lxqt/lxqt/issues/590#issuecomment-626878539)), and the other XDG portal implementations works (GTK and KDE).

**Comment by yan12125** [comment]
Hmm, not quite sure what's going on. Here is a wild guess: if you create a file `~/.config/environment.d/lxqt.conf` with the following content:
```
QT_QPA_PLATFORMTHEME=lxqt
```
And re-login or reboot, will things work? (Killing processes related to xdg-desktop-portal is not enough to make files under `environment.d` into effect)

This may be needed as under the current implementation, file dialogs are created by the process /usr/lib/xdg-desktop-portal-lxqt, and the latter is created by systemd (on Arch and most likely other systemd-based distros). However, systemd uses a different set of environment variables (defined by environment.d) than the shell (defined by `~/.bash_profile` or other init scripts) or LXQt components (defined by `$HOME/.config/lxqt/session.conf`).

**Comment by redtide** [comment]
For some reason I had to reboot to make it work, a re-login didn't. Thank you!

**Comment by tsujan** [comment]
> For some reason I had to reboot to make it work

Yes, I wanted to mention that. Sorry!

**Comment by redtide** [comment]
Isn't the case to create this environment variable setting, if not already present, in some installation step, LXQt or xdg-desktop-portal-lxqt?

**Comment by palinek** [comment]
FYI ... lxqt/xdg-desktop-portal-lxqt@52a3c9d

**Comment by yan12125** [comment]
Thanks! That fix looks clean. I think that means lxqt-qtplugin is now a runtime dependency of xdg-desktop-portal-lxqt. Could you add it to readme?

**Comment by palinek** [comment]
> Could you add it to readme?

lxqt/xdg-desktop-portal-lxqt@51b69e3

**Comment by redtide** [comment]
> Yeah the situation is less than ideal. Not sure if things can be improved.

Probably this is the wrong place to discuss it in this context but I'm facing also the problem that I can't uninstall it in any way on Arch, which to me it's a bug but I'm not sure if report in a new issue in the related repository (and reference this).

**Comment by stefonarch** [comment]
I compiled it directly and:
* on debian with only lxqt.portal in /usr/share/xdg-desktop-portal/portals/  it works just switching settings in firefox → about:config widget.use-xdg-desktop-portal to true
* on both arch installation with "true" no filedialog at all opens, unless desktop-portal-gtk or- kde is installed. Only way to make it work (until restart only) is installing xdg-desktop-portal-wlr and uninstalling it.
* 
This is with firefox not chrome, so maybe better open another issue?

**Comment by stefonarch** [comment]
The issue with arch is 
```
xdg-desktop-portal-impl 
(xdg-desktop-portal-gtk, xdg-desktop-portal-kde, xdg-desktop-portal-wlr, xdg-desktop-portal-gnome)
```
https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal/

There is no  way to install only `xdg-desktop-portal` afaik, so only workaround is 
* installing` xdg-desktop-portal xdg-desktop-portal-kde`
* removing  /usr/share/xdg-desktop-portal/portals/kde.portal
* reboot

**Comment by redtide** [comment]
I think we should open a new issue or at least discussion, we are offtopic here.
Other than that it's not true that it doesn't work on Arch, it works for me.

**Comment by tsujan** [comment]
Since I don't use portal, I can't comment on its problems. But it's obvious from @redtide's and @stefonarch's comments that there *is* a problem. So, please open a new issue an explain the problem(s) clearly.

**Comment by redtide** [comment]
NP, thank you @tsujan, I leave here for reference related [discussion](https://github.com/lxqt/lxqt/discussions/2123) and [issue](https://github.com/lxqt/xdg-desktop-portal-lxqt/issues/1).
